Sources/PackageModel/SwiftSDKBundle.swift

```diff
@@ -136,26 +136,35 @@ public struct SwiftSDKBundle {
         _ archiver: some Archiver,
         _ observabilityScope: ObservabilityScope
     ) async throws {
-        _ = try await withTemporaryDirectory(
-            removeTreeOnDeinit: true
-        ) { temporaryDirectory in
+        let bundleName = try await withTemporaryDirectory(removeTreeOnDeinit: true) { temporaryDirectory in
             let bundlePath: AbsolutePath
 
             if
                 let bundleURL = URL(string: bundlePathOrURL),
                 let scheme = bundleURL.scheme,
                 scheme == "http" || scheme == "https"
             {
-                let bundleName = bundleURL.lastPathComponent
+                let bundleName: String
+                let fileNameComponent = bundleURL.lastPathComponent
+                if fileNameComponent.hasSuffix(".tar.gz") {
+                    bundleName = fileNameComponent
+                } else {
+                    bundleName = "bundle.tar.gz"
+                }
                 let downloadedBundlePath = temporaryDirectory.appending(component: bundleName)
 
                 let client = HTTPClient()
-                var request = HTTPClientRequest.download(
+
+                var request = HTTPClientRequest.init(method: .head, url: bundleURL)
```

Comment:
```suggestion
                var request = HTTPClientRequest(method: .head, url: bundleURL)
```

Reply:
not sure I am reading this right. are we creating a head request, then immediately creating a download one?

Reply:
thanks, that's cleaned up now